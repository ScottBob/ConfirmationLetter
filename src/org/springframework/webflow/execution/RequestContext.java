package org.springframework.webflow.execution;

import com.oracle.webservices.internal.api.message.PropertySet;

import java.util.HashMap;
import java.util.Map;

public class RequestContext {
    public PropertySet getConversationScope() {
        PropertySet ps = new PropertySet() {
            HashMap<String, Object> stuff = new HashMap<>();

            @Override
            public boolean containsKey(Object o) {
                return stuff.containsKey(o);
            }

            @Override
            public Object get(Object o) {
                return stuff.get(o);
            }

            @Override
            public Object put(String s, Object o) {
                return stuff.put(s, o);
            }

            @Override
            public boolean supports(Object o) {
                return true;
            }

            @Override
            public Object remove(Object o) {
                return stuff.remove(o);
            }

            @Override
            public Map<String, Object> createMapView() {
                return stuff;
            }

            @Override
            public Map<String, Object> asMap() {
                return stuff;
            }
        };
        return ps;
    }
}
