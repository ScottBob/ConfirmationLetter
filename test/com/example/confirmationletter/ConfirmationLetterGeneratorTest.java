package com.example.confirmationletter;

import com.example.domain.*;
import com.example.record.command.FileUploadCommand;
import com.example.record.domain.FaultRecord;
import com.example.record.domain.Record;
import com.example.record.domain.TempRecord;
import com.example.record.parser.FileExtension;
import com.example.record.service.impl.Constants;
import org.approvaltests.Approvals;
import org.springframework.webflow.execution.RequestContext;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.HashMap;


public class ConfirmationLetterGeneratorTest {
    String[] creditDebit = {Constants.CREDIT, Constants.DEBIT};
    Currency[] currencies = {new Dollars(), new Euros(), new Pound(), new Yen()};


    @org.junit.Test
    public void letter() {
        ConfirmationLetterGenerator gen = new ConfirmationLetterGenerator();
        LetterSelector letterSelector = new LetterSelector();
        gen.setLetterSelector(letterSelector);
        RequestContext ctx = new RequestContext();
        FileUploadCommand fileUploadCommand = new FileUploadCommand();
        Client client = new Client();
        client.setCreditDebit(Constants.DEBIT);
        client.setAmountDivider(3.5);
        client.setCounterTransfer(Constants.FALSE);
        Profile profile = new Profile();

        client.setProfile(profile);
        HashBatchRecordsBalance hashBashRecordsBalance = new HashBatchRecordsBalance();
        hashBashRecordsBalance.setHashTotalCredit(80000.0);
        hashBashRecordsBalance.setHashTotalDebit(13549.11);
        hashBashRecordsBalance.setCollectionType("Fast");
        hashBashRecordsBalance.setTotalFee(49.85);
        hashBashRecordsBalance.setRecordsTotal(314.33);
        HashMap<Integer, BatchTotal> batchTotals = new HashMap<>();
        batchTotals.put(6,new BatchTotal(BigDecimal.valueOf(13), BigDecimal.valueOf(49)));
        batchTotals.put(27,new BatchTotal(BigDecimal.valueOf(2300), BigDecimal.valueOf(1699)));
        hashBashRecordsBalance.setBatchTotals(batchTotals);
        ArrayList<AmountAndRecordsPerBank> bankMap = new ArrayList<>();
        AmountAndRecordsPerBank anr = new AmountAndRecordsPerBank();
        anr.setBankName("Wells Fargo Bank");
        anr.setAccountNumber(31415);
        anr.setCurrencyType(Constants.CURRENCY_EURO);
        anr.setAmount(BigDecimal.valueOf(234532.23));
        anr.setTotalRecord(1);
        bankMap.add(anr);
        FaultRecord faultRecord = (FaultRecord)createRecord(true, "Wells Fargo", BigDecimal.valueOf(2342.24), "Captain America", 12, Constants.EUR_CURRENCY_CODE, Constants.DEBIT);
        ArrayList<FaultRecord> faultRecords = new ArrayList<>();
        faultRecords.add(faultRecord);
        FileExtension extension = new FileExtension(".txt");
        TempRecord record = (TempRecord) createRecord(false, "Chase Bank", BigDecimal.valueOf(19678.78), "Iron Man", 6, Constants.EUR_CURRENCY_CODE, Constants.DEBIT);
        ArrayList<Record> records = new ArrayList<>();
        records.add(record);
        TempRecord tempRecord1 = (TempRecord)createRecord(false, "Sun Life Bank", BigDecimal.valueOf(8733.19), "Black Widow", 3, Constants.FL_CURRENCY_CODE_FOR_WEIRD_BANK, Constants.CREDIT);
        ArrayList<TempRecord> faultyAccountNumberRecordsList = new ArrayList<>();
        faultyAccountNumberRecordsList.add(tempRecord1);
        TempRecord record1 = (TempRecord)createRecord(false, "Prosperity Bank", BigDecimal.valueOf(53247.67), "Wonder Woman", 7, Constants.FL_CURRENCY_CODE_FOR_WEIRD_BANK, Constants.CREDIT);
        ArrayList<TempRecord> sansDuplicateFaultRecordsList = new ArrayList<>();
        sansDuplicateFaultRecordsList.add(record1);
        TempRecord tempRecord2 = (TempRecord)createRecord(false, "Citibank", BigDecimal.valueOf(6733.19), "Ant Man", 3, Constants.FL_CURRENCY_CODE, Constants.DEBIT);
        records.add(tempRecord2);
        TempRecord record2 = (TempRecord)createRecord(false, "Bank of America", BigDecimal.valueOf(12247.67), "Batman", 7, Constants.USD_CURRENCY_CODE, Constants.DEBIT);
        records.add(record2);
        OurOwnByteArrayOutputStream stream = gen.letter(ctx,fileUploadCommand,client,hashBashRecordsBalance,"Houston Branch", bankMap, faultRecords, extension, records, faultyAccountNumberRecordsList, sansDuplicateFaultRecordsList);
        fileUploadCommand.setFee("no");
        OurOwnByteArrayOutputStream stream2 = gen.letter(ctx,fileUploadCommand,client,hashBashRecordsBalance,"Houston Branch", bankMap, faultRecords, extension, records, faultyAccountNumberRecordsList, sansDuplicateFaultRecordsList);
        Approvals.verify(stream.toString() + stream2.toString());
    }

    private Record createRecord(boolean faultRecord, String bankName, BigDecimal amount, String beneficiaryName, Integer beneficiaryAccountNumber, String currencyCode, String sign) {
        int isCounterTransferRecord = 1;
        Record record;
        if (faultRecord) {
            record = new FaultRecord();
        } else {
            record = new TempRecord();
        }
        record.setCurrencyCode(currencyCode);
        record.setSign(sign);
        Currency currency = new Euros();
        record.setCurrency(currency);
        record.setAmount(amount);
        Bank recordBank = new Bank();
        recordBank.setName(bankName);
        record.setBank(recordBank);
        record.setBeneficiaryName(beneficiaryName);
        record.setBeneficiaryAccountNumber(beneficiaryAccountNumber);
        record.setFreeRecord(2);
        record.setIsCounterTransferRecord(isCounterTransferRecord);
        return record;
    }
}