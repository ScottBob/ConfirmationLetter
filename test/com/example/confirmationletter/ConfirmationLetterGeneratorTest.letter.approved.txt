OurOwnByteArrayOutputStream{
creditDebit='-'
, confirmationLetter=ConfirmationLetter{
fileExtension=FileExtension{
extensionString='.txt'
}
, currency=null
, hashTotalCredit='80000.0'
, hashTotalDebit='13549.11'
, batchTotalDebit='0'
, batchTotalCredit='20'
, totalProcessedRecords='314.33'
, transactionCost='49.85'
, collectionType='Fast'
, bankMap=[AmountAndRecordsPerBank{bank=Wells Fargo Bank, totalRecord=1, amount=234532.23, currencyType='euro', accountNumber=31415, bankName='null'}]
, creditingErrors=[FaultRecord{
creditDebit='-'
, currencyCode='Euro'
, currency=Euros{
code='Euro'
, currencyType='null'
}
, bank=Wells Fargo
, accountNumber=12
, amount=2342.24
, freeRecord=2
, isCounterTransferRecord=1
, beneficiaryName='Captain America'
}]
, client=Client{creditDebit='-', profile=Profile{}, counterTransfer='0', amountDivider=3.5}
, branchName='Houston Branch'
, retrievedAmountEuro=19678.78
, retrievedAmountFL=6733.19
, retrievedAmountUSD=6733.19
, totalRecords=[]
}
}OurOwnByteArrayOutputStream{
creditDebit='-'
, confirmationLetter=ConfirmationLetter{
fileExtension=FileExtension{
extensionString='.txt'
}
, currency=null
, hashTotalCredit='80000.0'
, hashTotalDebit='13549.11'
, batchTotalDebit='0'
, batchTotalCredit='20'
, totalProcessedRecords='314.33'
, transactionCost=''
, collectionType='Fast'
, bankMap=[AmountAndRecordsPerBank{bank=Wells Fargo Bank, totalRecord=1, amount=234532.23, currencyType='euro', accountNumber=31415, bankName='null'}]
, creditingErrors=[FaultRecord{
creditDebit='-'
, currencyCode='Euro'
, currency=Euros{
code='Euro'
, currencyType='null'
}
, bank=Wells Fargo
, accountNumber=12
, amount=2342.24
, freeRecord=2
, isCounterTransferRecord=1
, beneficiaryName='Captain America'
}]
, client=Client{creditDebit='-', profile=Profile{}, counterTransfer='0', amountDivider=3.5}
, branchName='Houston Branch'
, retrievedAmountEuro=19678.78
, retrievedAmountFL=6733.19
, retrievedAmountUSD=6733.19
, totalRecords=[]
}
}